app/src/main/java/com/aaronjwood/portauthority/async/ScanHostsAsyncTask.java

```diff
@@ -228,16 +259,43 @@ protected final void onProgressUpdate(Void... params) {
                         activity1.processFinish(e);
                         return;
                     }
+                    /**
+                    * BUG: Some devices don't respond to mDNS if NetBIOS is queried first. Why?
+                    * So let's query mDNS first, to keep in mind for eventual UPnP implementation.
+                    **/
+                    try {
+                        MDNSResolver resolver = new MDNSResolver(UserPreference.getLanSocketTimeout(ctx));
+                        InetAddress add = InetAddress.getByName(ip);
+                        String name = resolver.resolve(add);
+                        resolver.close();
+                        if (name != null && !name.isEmpty()) {
+                            host.setHostname(name);
+                            if (activity1 != null) {
+                                // Call with null to refresh
+                                activity1.processFinish(null, numHosts);
+                            }
+                            return;
+                        }
+                    }
+                    catch (Exception e) {
+                        // It's common that many discovered hosts won't have a mDNS entry.
+                    }
 
                     try {
-                        NbtAddress[] netbios = NbtAddress.getAllByAddress(ip);
-                        for (NbtAddress addr : netbios) {
-                            if (addr.getNameType() == NETBIOS_FILE_SERVER) {
-                                host.setHostname(addr.getHostName());
-                                return;
+                        NetBIOSResolver resolver = new NetBIOSResolver(UserPreference.getLanSocketTimeout(ctx));
```

Comment:
Can we reuse the code from above here too? It looks like the resolver assignment is the only thing that's different.